### ISSUE TITLE
[schema-customization] fields defined with sanitzed names don't resolve to unsanitized name in source data

### ISSUE THREAD

**Issue by pieh** [gatsbyjs/gatsby#14102]
## Description

If node data has field name that can't be used in graphql (i.e. containing `?`, `(`, `)` characters etc), there is no quick way to define type without explicitly adding resolver that will pick from source using unsanitized field name. This is regression / behaviour change from gatsby@<2.5.0

### Steps to reproduce

https://github.com/pieh/schema-250-sanitized-scalars

install deps -> gatsby build

See https://github.com/pieh/schema-250-sanitized-scalars/blob/master/gatsby-node.js (mostly type definition - I added some information in comments)

### Expected result

Not sure about expected result, but previous behaviour was that using
```
field_that_needs_to_be_sanitized_: "String"
```
for field definition would ultimately result in actually using un-sanitized field name (probably because resolver was inferred?). From PR description and blog post I'm getting that there should be no behaviour changes.

### Actual result

Console output will show:
```
Results
{ field_that_needs_to_be_sanitized_: null,
  _another__field_that_needs_to_be_sanitized: 'bar',
  _third_field_that_needs_to_be_sanitized: 'baz' }

Expected results
{ field_that_needs_to_be_sanitized_: 'foo',
  _another__field_that_needs_to_be_sanitized: 'bar',
  _third_field_that_needs_to_be_sanitized: 'baz' }

"field_that_needs_to_be_sanitized_" field resolves with "null" instead of value of "source['field_that_needs_to_be_sanitized?']"
```

### Environment

```
  System:
    OS: macOS 10.14
    CPU: (12) x64 Intel(R) Core(TM) i9-8950HK CPU @ 2.90GHz
    Shell: 5.3 - /bin/zsh
  Binaries:
    Node: 10.9.0 - /usr/local/bin/node
    Yarn: 1.10.1 - /usr/local/bin/yarn
    npm: 6.9.0 - /usr/local/bin/npm
  Languages:
    Python: 2.7.10 - /usr/bin/python
  Browsers:
    Firefox: 66.0.4
    Safari: 12.0
  npmPackages:
    gatsby: ^2.5.0 => 2.5.0 
  npmGlobalPackages:
    gatsby-cli: 2.5.3
    gatsby-dev-cli: 2.4.14
```

Notes:
 - if this will be addressed - revert https://github.com/gatsbyjs/gatsby/pull/14096

/cc @freiksenet

**Comment by freiksenet** [comment]
Fixed in `2.15.1`.

**Comment by m-allanson** [comment]
This was fixed in 2.5.2 and #14096 has been reverted!
